core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,396 @@
+          Caffeine.newBuilder()
+              .build(
+                  specId -> {
+                    PartitionSpec spec = transformedSpecs.get(specId);
+                    Expression filter =
+                        context().ignoreResiduals() ? Expressions.alwaysTrue() : filter();
+                    return ResidualEvaluator.of(spec, filter, isCaseSensitive());
+                  });
+      LoadingCache<Integer, String> specStringCache =
+          Caffeine.newBuilder()
+              .build(
+                  specId -> {
+                    PartitionSpec spec = transformedSpecs.get(specId);
+                    return PartitionSpecParser.toJson(spec);
+                  });
+
+      // iterate through delete manifests
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(tableOps().io()));
+      CloseableIterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              deleteManifests,
+              manifest -> {
+                // Filter partitions
+                CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
+                    ManifestFiles.readDeleteManifest(manifest, tableOps().io(), transformedSpecs)
+                        .caseSensitive(isCaseSensitive())
+                        .select(scanColumns())
+                        .filterRows(rowFilter)
+                        .liveEntries();
+
+                // Filter delete file type
+                CloseableIterable<ManifestEntry<DeleteFile>> positionDeleteEntries =
+                    CloseableIterable.filter(
+                        deleteFileEntries,
+                        entry -> entry.file().content().equals(FileContent.POSITION_DELETES));
+
+                Types.StructType partitionType = Partitioning.partitionType(table());
+
+                return CloseableIterable.transform(
+                    positionDeleteEntries,
+                    entry -> {
+                      int specId = entry.file().specId();
+                      return new PositionDeleteScanTaskImpl(
+                          entry.file().copyWithoutStats(),
+                          schemaString,
+                          specStringCache.get(specId),
+                          residualCache.get(specId),
+                          partitionType);
+                    });
+              });
+
+      return new ParallelIterable<>(results, planExecutor());
+    }
+  }
+
+  /** Scan task for position delete files */
+  public interface PositionDeleteScanTask extends ContentScanTask<DeleteFile> {
+
+    /**
+     * Utility method to get constant values of rows to be scanned by this task. The following
+     * columns are constants of each PositionDeletesFileScanTask: * spec_id * partition *
+     * delete_file_path
+     *
+     * @return a map of column id to constant values returned by this task type
+     */
+    Map<Integer, ?> constantsMap();
```

Comment:
Are we sure this should be part of the task API? It feels that a utility (used in the reader) would make more sense. That utility could accept this task and then will have access to the file, spec and partition tuple to build these values.

Reply:
This would be in line with `PartitionUtil$constantsMap`.

Reply:
Yea didn't want to duplicate the code.  Took your earlier suggestion and just used the existing MetadataColumn ids so we can re-use PartitionUtil as is when we implement the spark read side.  However, this table still declares them not not as metadata columns per-se, but as declared.

Reply:
Actually , reverted all of them to non-hidden column names, but keep metadata column id.

Reply:
Makes sense.